lib/src/space/mod.rs

```diff
@@ -196,9 +202,16 @@ pub trait SpaceMut: Space {
     fn as_space(&self) -> &dyn Space;
 }
 
-use crate::common::shared::Shared;
+#[macro_export]
+macro_rules! space_box {
```

Comment:
Could we use specific `SpaceBox` constructor instead of macros? Always using macros to construct an instance of the type seems a bit non-obvious.

Reply:
Done in 9fe5994